docs/querying/functions.md

```diff
@@ -401,6 +401,14 @@ in ascending order.
 
 Same as `sort`, but sorts in descending order.
 
+## `sort_by_label()`
+
+`sort_by_label(v instant-vector, label string, ...)` returns vector elements sorted by their label values and sample value in case of label values being equal, in ascending order.
```

Comment:
Hmm, I wonder if we shouldn't keep this completely independent from the sample value and instead use other label values to break the sorting tie? That way, sort results would be stable over time, given the same set of time series.

Reply:
I agree with Julius. We could also decide not to sort at all in that case, so you can do sort_by_label(sort(up),"job")

Reply:
Yeah, that sounds even better.

Reply:
So I should remove the "fallback" to comparing the value in the `sort_by_label` functions? This might make the test unstable, though I have to re-check if that happens without this "fallback" compare.

Reply:
The tests should be stable even without this

Reply:
we should make it clear that labels are sorted lexicographically

Reply:
@roidelapluie Without the value comparison the test fails no matter in what order the result in the test(s) are placed.

Reply:
That is strange. The order should be predictible.

Reply:
It seems that it is not predictable as every X run the test fails and complains about the order of the result.

How should we approach this that we can hopefully get this functionality integrated?

Reply:
From my perspective, the fix here is to sort the values using the `sort`/`sort_desc` functions and then do the sort by label(s).